servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -785,11 +799,18 @@ public void mkdirs(TachyonURI path, boolean recursive) throws InvalidPathExcepti
     synchronized (mInodeTree) {
       try {
         List<Inode> created = mInodeTree.createPath(path, 0, recursive, true);
-        // Journal all of the created inodes.
-        for (Inode inode : created) {
-          writeJournalEntry(inode.toJournalEntry());
-        }
+
         writeJournalEntry(mDirectoryIdGenerator.toJournalEntry());
+        for (int i = 0; i < created.size(); i ++) {
```

Comment:
The logic is identical to the one in createFile. It might be worth extracting it into a private function.